src/bindings/cpp/tests/testcpp_key.cpp

```diff
@@ -13,6 +13,52 @@
 #include <string>
 #include <vector>
 
+/* check if exceptions are thrown if the underlying c-functions return error codes (-1 or NULL) */
+TEST (key, cErrors)
+{
+	Key k ("user:/key", KEY_VALUE, "testkey", KEY_END);
+	Key x = nullptr;
+
+	EXPECT_THROW (x.addName (""), KeyInvalidName);
+	EXPECT_THROW (x.setName (""), KeyInvalidName);
+	EXPECT_THROW (x.addBaseName (""), KeyInvalidName);
```

Comment:
The test is confusing. `addBaseName ("")` would succeed, it is only about that x == nullptr.

Reply:
Thanks for the feedback!

I now created new separated tests for normal keys, null-keys, keys added to a KeySet and metakeys and added them to the respective test suites. (testcpp_key, testcpp_ks, testcpp_meta)
I hope things are clearer and more comprehensive now.